test-utils/src/random.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crypto::random::{Rng, SeedableRng};
+use rand_chacha::ChaChaRng;
+
+#[derive(Debug, Copy, Clone)]
+pub struct Seed(pub u64);
+
+impl Seed {
+    pub fn from_entropy() -> Self {
+        Seed(crypto::random::make_true_rng().gen::<u64>())
+    }
+
+    pub fn from_u64(v: u64) -> Self {
+        Seed(v)
+    }
+}
+
+#[must_use]
+pub fn make_seedable_rng(seed: Seed) -> impl Rng {
+    ChaChaRng::seed_from_u64(seed.0)
+}
```

Comment:
You have narrowed down the seed to 64 bits but still use a PRNG which has, I believe a much larger internal state. Is that intended?

Reply:
The reason I chose ChaCha is that as far as I understood it is more stable across the platform and over time, which might not be the case with StdRng. But not sure how important it is for us.

Reply:
> The reason I chose ChaCha is that as far as I understood it is more stable across the platform and over time, which might not be the case with StdRng. But not sure how important it is for us.

That sounds like a very useful thing to have especially if the goal is test reproducibility.

Reply:
It was me that suggested something smaller... because u256 is huge and unnecessary and harder to read. If we run the tests 1000 times per second, for 100 years, we won't finish a millionth of u64 range... so there's practically no point in using anything larger than u64.